channel/channel.go

```diff
@@ -0,0 +1,103 @@
+package channel
+
+import (
+	"context"
+	"errors"
+	"runtime/debug"
+	"time"
+)
+
+type TimedChannel[T any] struct {
+	t      time.Duration
+	ctx    context.Context
+	c      chan T
+	pushCb func(string)
+	popCb  func(string)
```

Comment:
I'm not sure what's the use of these callbacks.
Why would we want to call a callback with the stack trace ?
I'm a bit dumbfounded.